From c5fe437ba5670ded381047ab6036eaf11eaa1d2d Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Mon, 28 Apr 2014 15:35:35 +0200
Subject: [PATCH 19/28] Extractor: Check if the file should be indexed

The check is a file should be indexed was intitally only performed
during the first phase. With this we also do it in the second phase.

This is useful as if the cleaner stops running (killed, crashed, system
shut down) it does to respawn itself and stuff continued to get indexed.
With this the extractor will clean up all the files which should not be
indexed.
---
 src/file/extractor/CMakeLists.txt | 3 +++
 src/file/extractor/app.cpp        | 7 +++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/file/extractor/CMakeLists.txt b/src/file/extractor/CMakeLists.txt
index baaf571..df67af5 100644
--- a/src/file/extractor/CMakeLists.txt
+++ b/src/file/extractor/CMakeLists.txt
@@ -6,6 +6,9 @@ set(EXTRACTOR_SRCS
   ../basicindexingjob.cpp
   ../database.cpp
   ../util.cpp
+  ../fileindexerconfig.cpp
+  ../regexpcache.cpp
+  ../fileexcludefilters.cpp
 )
 
 kde4_add_executable(baloo_file_extractor ${EXTRACTOR_SRCS})
diff --git a/src/file/extractor/app.cpp b/src/file/extractor/app.cpp
index 9c7a70f..970de1b 100644
--- a/src/file/extractor/app.cpp
+++ b/src/file/extractor/app.cpp
@@ -24,6 +24,7 @@
 #include "../basicindexingjob.h"
 #include "../database.h"
 #include "xapiandatabase.h"
+#include "../fileindexerconfig.h"
 
 #include <KCmdLineArgs>
 #include <KMimeType>
@@ -64,6 +65,8 @@ App::App(QObject* parent)
     m_bData = args->isSet("bdata");
     m_debugEnabled = args->isSet("debug");
 
+    FileIndexerConfig config;
+
     m_results.reserve(args->count());
     for (int i=0; i<args->count(); i++) {
         FileMapping mapping = FileMapping(args->arg(i).toUInt());
@@ -81,11 +84,11 @@ App::App(QObject* parent)
             url = args->url(i).toLocalFile();
         }
 
-        if (QFile::exists(url)) {
+        if (QFile::exists(url) && config.shouldBeIndexed(url)) {
             m_urls << url;
         } else {
             // id or url was looked up, but file deleted
-            kDebug() << url << "does not exist";
+            kDebug() << url << "does not exist or should not be indexed";
 
             // Try to delete it as an id:
             // it may have been deleted from the FileMapping db as well.
-- 
1.9.0

